### PR TITLE
Add getAccountMode() and getCurrentAccount() to MSAuthService

### PR DESCRIPTION
**Pull request by rpdome**
Introducing 2 more functions to MicrosoftAuthService.
- getAccountMode()
- getCurrentAccount()

Please see [the spec](https://identitydivision.visualstudio.com/IDDP/_git/general-docs/pullrequest/744?_a=files&path=%2FSharedDevice%2FSharedDeviceAndroid.md) for reference.

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java [hunk 1]
```diff
@@ -1018,6 +1018,25 @@ private AuthenticationConstants() {
          */
         public static final String BROKER_RESULT_V2 = "broker.result.v2";
 
+        /**
+         * String to return the broker account mode.
+         * This is used to determine what PublicClientApplication MSAL will return to its caller.
+         */
+        public static final String BROKER_ACCOUNT_MODE = "broker.account.mode";
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/AuthenticationConstants.java]: We recently switched to make these broker constant values snake case (changes in release branch). Can we do the same here for all new constants like `broker.account.mode` to `broker_account_mode

File: common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java [hunk 2]
```diff
@@ -417,4 +419,62 @@ public Bundle bundleFromAccountRecordList(@NonNull final List<AccountRecord> rec
...
+        resultBundle.putString(BROKER_ACCOUNT_MODE, accountMode);
+        return resultBundle;
+    }
+
+    /**
+     * Get Account mode from bundle.
+     * @param bundle Bundle
+     * @return Account mode.
+     */
+    public static String getAccountModeFromBundle(@NonNull final Bundle bundle) {
```
Comment (kreedula) [inline common/src/main/java/com/microsoft/identity/common/internal/result/MsalBrokerResultAdapter.java]: Rename to `accountModeFromBundle` for method naming consistency in this class

**Comment by rpdome** [comment]
force pushed as I accidentally add the changes of another PR into this one.

File: common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl [hunk 3]
```diff
@@ -32,8 +32,12 @@ interface IMicrosoftAuthService {
 
     Bundle hello(in Bundle bundle);
 
+    Bundle getAccountMode();
```
Comment (heidijinxujia) [inline common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl]: Question, who is setting the account mode? Broker or MSAL application?

File: common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl [hunk 4]
```diff
@@ -32,8 +32,12 @@ interface IMicrosoftAuthService {
...
+
     Bundle getAccounts(in Bundle bundle);
 
+    Bundle getCurrentAccount();
```
Comment (heidijinxujia) [inline common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl]: 1. Should the getCurrentAccount() only be called when the account mode is "single"?
2.  If not, what should it return when multiple accounts exist? 
3. What kind of accounts are returning here? I assume WPJ account only, right?

Comment (rpdome) [reply common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl]: 1. Yes
2. An error bundle.
3. an account that holds BRT. In the shared device mode, there will be only 0 or 1 account that holds BRT, and that account MUST not be the WPJ account. (The WPJ account is used by the admin to bootstrap the WPJ/shared device state only)

File: common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl (hunk 3 above)
Comment (rpdome) [reply common/src/main/aidl/com/microsoft/identity/client/IMicrosoftAuthService.aidl]: Broker. 

Here's roughly what happens in FLW scenario
1. Admin bootstrap a device with their 'admin' WPJ account through Authenticator app.
2. WPJ flows sends the flag to the server side, the server validates WPJ account's role.
3. If the role matches, the WPJ operation succeeds and the device is marked as shared - therefore a 'mode' is set.
4. MSAL queries this 'mode' to determine which set of Broker functionality it should expose to MSAL application.

**Pull request merged**